### ISSUE TITLE
Discuss first release of the UI project

### ISSUE THREAD

**Issue by vishnoianil** [instructlab/ui#8]
We need to address few major questions to drive the community focus.

- what all features do we want to include in our first release?
- approximate timeline for the first release?
- where should we host the first release?
- do we want to enable any cost control mechanism? 

If you have more questions that we should discuss in this context, please feel free to the list.

**Comment by renrenstart** [comment]
The InstructLab Insights web page can be part of the first release.

Currently the web page is static html file wrapped by react framework along with github oauth.

It'll be great to have the web page re-written in react and included as part of the main UI app. We can also add a report list page in the main app.

We can discuss the above in our next meeting with @juancappi Nitin.

**Comment by renrenstart** [comment]
@nitin-ramchandani

**Comment by toraponibm** [comment]
Here are some thoughts on initial functions for contributors.

1. Ability to create a contribution
- Enter contribution (skill & knowledge) contents on UI (without writing YAML)
- Validate mandatory fields, for skills and knowledges each
- Validate content format (such as URL path, minimum length, etc.)

2. Ability to view current model outputs while creating seed examples
- View response(s) from existing models for a question text.

3. Ability to search existing taxonomy
- Search by keywords
- Advanced search (e.g. similar search) is TBD

4. Ability to generate YAML
- Generate valid and formatted YAML that confirms to ilab checks

5. Ability to interact with GitHub
- Submit contribution as new PR
- Update existing PR
- View insights
- View the status and progress of the user contributions as a list

6. Ability to interact with triagers
- TBD

**Comment by Gregory-Pereira** [comment]
To recap this with the state of our current release readiness, and my less than organized thoughts on this topic... :

- [ ]  Ability to create a contribution
    - [x] Enter contribution (skill & knowledge) contents on UI (without writing YAML) 
    - [x] Validate mandatory fields, for skills and knowledges each (although validation is just exists or not, no email validation for example)
    - [ ] Validate content format (such as URL path, minimum length, etc.)
- [ ] Ability to view current model outputs while creating seed examples
    - [~] Task: View response(s) from existing models for a question text. 
    - Response: This is not actual as trivial as it sounds. What we can currently do: Chat with the model and ask it questions, and it can use context from responses. What we cannot do is start chatting with a model, train with QNA and then chat with it afterward. In other words, at this point chatting with the model is solely for identifying gaps in its knowledge. Not sure if this addresses your original conception of this point but I wanted to make the status clear.
- Ability to search existing taxonomy
   - [ ] Task: We currently have no ability to interact with the taxonomy.
   - Response: However, according to legend, there is a mysterious taxonomy viewer roaming around somewhere. Have never seen a demo of it or talked to someone who has directly used it in my ~6 weeks with the taxonomy team but apparently it exists, so ¯\_(ツ)_/¯ . However, I do not this should be included in v1 of the UI, if it even gets included at all. Discussion around taxonomy structure and storing knowledge at scale belongs to the taxonomy team, and this is meant more so a general purpose tool for making contributions, I would argue that what you are describing is more taxonomy lifecycle which is a different concept.
- [x] Ability to generate YAML
   - [~] Generate valid and formatted YAML that confirms to ilab checks. Pretty sure we don't do the ilab-checks, but for v1 I suggest we just leverage the native checks in the submission process with just basic standard yaml validation.
- [~] Ability to interact with GitHub
    - [x] Submit contribution as new PR
    - [x] Update existing PR
    - [ ] View insights - no current insights view but I would argue this is v2. 
    - [ ] View the status and progress of the user contributions as a list - again I think this is v2
- [ ] Ability to interact with triagers
    - It was decided in taxonomy + UI call that we should start here by leveraging the experience of the triage team. We should scope v1 of the UI to JUST being making a contribution via Github, show the user where it is there, and expect the triage team to work with them to resolve all their feedback. This way we get the tool out there but still can expect minimal if any Github knowledge by relying on the Triage team. I would rather have scale issues here and have to change our approach than delay release to handle a scaling issue we dont have yet.
    
In summary: @toraponibm I think you spent a LOT of thought on this, and these are bloody amazing ideas. It might just me my personal eagerness to showoff this great work, but I think we should refine scope as much as possible for the v1 release and iterate from there. It should be make a contribution and expect the triage team to do the leg work of getting it into a submittable state from little to no feedback over Github, and even though its pushing this responsibility onto them, it is - at least to some degree - in their job description for now.

TLDR: Ship fast, small and early
